tests/baselines/reference/bestCommonTypeOfTuple2.types

```diff
@@ -29,7 +29,7 @@ class F extends C { f }
 class C1 implements base1 { i = "foo"; c }
 >C1 : C1
 >base1 : base1
->i : string
+>i : any

```

Comment:
After lots of discussion with you and @RyanCavanaugh, I propose a more complex rule that would revert this test change [1]:
1. If a property has a type annotation, use that.
2. If a property has an inherited type and **no** initialiser, use that.
3. If a property has an inherited type and an initialiser, contextually type the initialiser with the inherited type, then use the type of the initialiser.

This makes the following examples correct:

```ts
interface Comparer {
  compare(x: any, y: any): number;
}
class NumCmp implements Comparer {
  compare = (x: number, y: number) => x - y;
}
let cmp = new NumCmp();
cmp.compare("foo", "bar") // ERROR here

interface Long {
  length: number;
}
class Cat implements Long {
  length = undefined;
}
let longcat = new Cat();
longcat.length = "error here"; // ERROR
```

[1] Note that the behaviour that caused the change is technically correct but it is a breaking change.